### DISCUSSION TITLE
Server-Profiles and Facets

### DISCUSSION THREAD

**Discussion by AndreasHeine** [FreeOpcUa/opcua-asyncio#434] [Ideas]
Hey,

just a quick survey:

at the moment the serverclass has always all currently implemented facets and features. 
the opcua spec. defines some server-profiles which is a set of features a server has to offer:

> "Nano Embedded Device 2017 Server Profile" Profile
> "Micro Embedded Device 2017 Server Profile" Profile
> "Embedded 2017 UA Server Profile" Profile
> "Standard 2017 UA Server Profile" Profile

https://apps.opcfoundation.org/profilereporting/

in order to support profiles we would need to split the serverclass into different inheriting serverclasses
```
class NanoServer(objects)
class MicroServer(Nano)
...
```

**this would be a breaking change!**

i am asking you all if we should make such a big step which would change the server-architecture of this library

looking forward to your feedback :)

**Comment by oroulet** [comment]
This is not going to break so much. Is it? you just split the big InternalServer and maybe InternalSession in some classes inheriting each others. Very few people use these classes directly. And maybe the highe level Server class can take an argument to specify what profile to support
